Source/1.4/ShipInteriorMod2.cs

```diff
@@ -911,6 +953,21 @@ public static HashSet<IntVec3> FindAreaAttached(Building root, bool includeRock
 			}
 			return cellsFound;
 		}
+
+		public static bool IsRock(TerrainDef def)
+		{
+			return def == TerrainDef.Named("Slate_Rough")
+				|| def == TerrainDef.Named("Slate_RoughHewn")
+				|| def == TerrainDef.Named("Marble_Rough")
+				|| def == TerrainDef.Named("Marble_RoughHewn")
+				|| def == TerrainDef.Named("Granite_Rough")
+				|| def == TerrainDef.Named("Granite_RoughHewn");
+		}
```

Comment:
private const string [] RockTerrains = { "Slate_Rough", "Slate_RoughHewn", "Marble_Rough", "Marble_RoughHewn", "Granite_Rough", "Granite_RoughHewn" };

public static bool IsRock(TerrainDef def)
{
  return Array.Exists(RockTerrains, name => def == TerrainDef.Named(name));
}

Reply:
Fixed. I've cached TerrainDef instead. Though, according to my measurements:

```
prepare=14.606ms
processSourceArea=131.7375ms
destroySource=85.0549ms
moveThings=5162.8475ms
moveZones=0.0158ms
moveTerrain=0.4849ms
moveRoof=163.2749ms
finalize=35.5809ms
```
All calls to `IsRock` are done inside *processsourceArea* section of MoveShip. For now it is far from being a bottleneck.